test/unit/provisioning/device-group-utils_test.js

```diff
@@ -0,0 +1,172 @@
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * fiware-iotagent-lib is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with fiware-iotagent-lib.
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[contacto@tid.es]
+ */
+'use strict';
+
+/* jshint camelcase: false */
+
+var iotAgentLib = require('../../../lib/fiware-iotagent-lib'),
+    utils = require('../../tools/utils'),
+    should = require('should'),
+    async = require('async'),
+    groupRegistryMemory = require('../../../lib/services/groups/groupRegistryMemory'),
+    request = require('request'),
+    groupCreation = {
+        url: 'http://localhost:4041/iot/services',
+        method: 'POST',
+        json: utils.readExampleFile('./test/unit/examples/groupProvisioningRequests/provisionFullGroup.json'),
+        headers: {
+            'fiware-service': 'TestService',
+            'fiware-servicepath': '/testingPath'
+        }
+    },
+    alternateGroupCreation = {
+        url: 'http://localhost:4041/iot/services',
+        method: 'POST',
+        json: utils.readExampleFile('./test/unit/examples/groupProvisioningRequests/provisionFullGroupAlternate.json'),
+        headers: {
+            'fiware-service': 'TestService',
+            'fiware-servicepath': '/testingPath'
+        }
+    },
+    iotAgentConfig = {
+        logLevel: 'FATAL',
+        contextBroker: {
+            host: '192.168.1.1',
+            port: '1026'
+        },
+        server: {
+            port: 4041,
+            baseRoot: '/'
+        },
+        types: {
+            'Termometer': {
+                commands: [],
+                lazy: [
+                    {
+                        name: 'temp',
+                        type: 'kelvin'
+                    }
+                ],
+                active: [
+                ],
+                apikey: '1234567890asdfghjkl',
+                service: 'TestService',
+                subservice: '/testingPath'
+            }
+        },
+        service: 'smartGondor',
+        subservice: 'gardens',
+        providerUrl: 'http://smartGondor.com',
+        deviceRegistrationDuration: 'P1M',
+        throttling: 'PT5S',
+        defaultKey: 'default1234'
+    };
+
+describe('Device Group utils', function() {
+    afterEach(function(done) {
+        iotAgentLib.deactivate(function() {
+            groupRegistryMemory.clear(done);
+        });
+    });
+
+    describe('When an APIKey is requested for a device in a group without the SingleConfiguration mode', function() {
```

Comment:
There are many appearances of `APIKey` all together in this file, I do not know if you want to keep it that way of changing it to `API Key`... :)